src/QirRuntime/experimental/CircuitToJsonPrinter.cpp

```diff
@@ -0,0 +1,659 @@
+        {
+            this->owner->R(axis, qubit, CalculateAngle(numerator, power));
+        }
+
+        void ControlledRFraction(
+            long numControls,
+            Qubit* const controls,
+            PauliId axis,
+            Qubit qubit,
+            long numerator,
+            int power) override
+        {
+            this->owner->ControlledR(numControls, controls, axis, qubit, CalculateAngle(numerator, power));
+        }
+
+        void R1Fraction(Qubit qubit, long numerator, int power) override
+        {
+            this->owner->R1(qubit, CalculateAngle(numerator, power));
+        }
+
+        void ControlledR1Fraction(long numControls, Qubit* const controls, Qubit qubit, long numerator, int power)
+            override
+        {
+            this->owner->ControlledR1(numControls, controls, qubit, CalculateAngle(numerator, power));
+        }
+
+        // TODO: implement exponents
+    };
+
+    /*==============================================================================
+        CCircuitToJsonTranslator
+    ==============================================================================*/
+    class CCircuitToJsonTranslator final : public ITranslator
+    {
+        // List of the instructions to be printed that corresponds to the commands executed by the translator.
+        vector<Instruction> instructions;
+
+        // Errors, encountered by the translator (might or might not be ignorable).
+        vector<TranslationError> errors;
+
+        // To track the width of the circuit.
+        long quantumRegisterSize = 0;
+
+      public:
+        CCircuitToJsonTranslator() = default;
+        ~CCircuitToJsonTranslator() = default;
+
+        void AddInstruction(Instruction&& instr)
+        {
+            if (instr.gate == Gate::alloc)
+            {
+                this->quantumRegisterSize++;
+            }
+            else
+            {
+                instructions.push_back(std::move(instr));
+            }
+        }
+
+        void AddError(TranslationError&& error)
+        {
+            this->errors.push_back(std::move(error));
+        }
+
+        void PrintRepresentation(ostringstream& os, ostringstream* err) override
+        {
+            os << "{\"qubits\":" << this->quantumRegisterSize << ",\"circuit\":[";
+            for (const Instruction& inst : this->instructions)
+            {
+                inst.Print(os);
+            }
+            os << "]}";
+
+            if (err != nullptr && !this->errors.empty())
+            {
+                *err << "{\"error_count\":" << this->errors.size() << ",\"errors\":[";
+                for (const TranslationError& error : this->errors)
+                {
+                    error.Print(*err);
+                }
+                *err << "]}";
+            }
+        }
+    };
+
+    /*==============================================================================
+        CCircuitPrintingSimulator

```

Comment:
It's a common readability practice in native code bases.

Reply:
Following our wiki regarding C++ style rules, it doesn't seem to be encouraged: https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#S-class
The practice seems related to the general use of Hungarian notation and based on a quick search it sounds like it may be common mostly in certain older code bases. Given that our native simulator doesn't follow that practice, we should make a call in general, and I would suggest to follow the page our wiki leads to and drop the C. I am happy to set up a broader decision for that, to document this call if you feel this is likely something that other people might wonder about.